src/HealthChecks.Elasticsearch/ElasticsearchHealthCheck.cs

```diff
@@ -21,9 +21,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
     {
         try
         {
-            if (!_connections.TryGetValue(_options.Uri, out var lowLevelClient))
+            if (!_connections.TryGetValue(_options.Uri, out var elasticsearchClient))

```

Comment:
not related to your changes, but something that we should improve in the future:

From https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/recommendations.html:

> Client reuse can be achieved by creating a singleton static instance or by registering the type with a singleton lifetime when using dependency injection containers.

So same as https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/2113 we should try to steer the users towards best practices and provide and recommend overloads that accept `ElasticsearchClient` and simply re-use it (to there would be one instance for the whole app, not one for health checks and another for the app itself).

Reply:
what about many health checks for different Elastic instances?

Reply:
@adamsitnik, is the last change providing all the things we need?
I think we should make all the changes here.